projects/plugins/jetpack/extensions/blocks/cookie-consent/edit.js

```diff
@@ -29,10 +139,86 @@ function CookieConsentBlockEdit( { attributes, setAttributes } ) {
 		} );
 	}
 
+	async function goToTemplatePart( savedTemplate ) {
+		setIsSaving( false );
+		removeBlock( clientId );
+		window.open(
+			`/wp-admin/site-editor.php?postType=${ savedTemplate.type }&postId=${ savedTemplate.id }`
+		);
+	}
+
 	const blockProps = useBlockProps( {
 		className: `wp-block-jetpack-cookie-consent align${ align }`,
 	} );
 
+	const isInWarningState = parentPost?.slug !== PART_SLUG;
+
+	/* If the block is added in the right place (in its own part), mark it as such, this is needed in the save function */
+	useEffect( () => {
+		if ( parentPost ) {
+			setAttributes( { isInWarningState } );
+		}
+	}, [ setAttributes, parentPost, isInWarningState ] );
+
+	/* If the block is added in the wrong place (not in its own part), render UI that helps the user create a template part. */
+	if ( isInWarningState ) {
+		if ( mode === 'LOADING' ) {
+			return __( 'Loading…', 'jetpack' );
+		} else if ( mode === 'PART_EXISTS' ) {
+			return (
+				<Card>
+					<CardBody>
+						<p>
+							{ __(
+								'In order to be rendered on every page of your site, the Cookie Consent Block is best added as a template part. You already have created a template part before.',
```

Comment:
```suggestion
								'You can only have one cookie consent banner on your site. To edit the one you already have click the button below.',
```

This is a bit more informative for me, but I could be convinced otherwise.

Reply:
Ohh I love this one!